### PR TITLE
New docker Mac OS X and Windows don't relay on docker-machine

### PR DESCRIPTION
**Pull request by dzwicker**
New docker Mac OS X and Windows versions don't depend on docker-machine any more. So we can remove the DockerSetupTask. See #36

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -41,23 +41,37 @@ if (System.env.CIRCLE_TEST_REPORTS) {
...
             from components.java
             artifact(sourceJar)
             artifact(publishPluginJavaDocsJar)
         }
     }
+    repositories {
+        maven {
+            credentials {
+                username 'deployment'
+                password 'MXJmqxvTGve9s6q9y7Ks'
```
Comment (markelliot) [inline build.gradle:68]: pretty sure you don't want to commit the set of changes to this file, and you might want to think about changing this exposed password.

**Comment by markelliot** [comment]
Given that the Docker for Mac isn't widely available yet, I'm not sure it's safe to make this change. I'd be open to having an extra configuration item in the extension to enable/disable the setup task, though.

**Comment by jkozlowski** [comment]
I hit the same problem, as a workaround running with -x dockerSetup works fine for me. In general I think it'd be better if we ran something like docker ps and checked return code to see if we can connect to docker: that would potentially work with docker-machine and docker for mac beta.

**Comment by markelliot** [comment]
I think test running docker ps makes sense, happy to see that in a PR.

**Comment by SerialVelocity** [comment]
[dlite](https://github.com/nlf/dlite) also fails to work with this too.

**Comment by uschi2000** [comment]
Docker for Mac is now a public beta, so let's make some progress here. @jkozlowski will you drive this or should I?

**Comment by jkozlowski** [comment]
see #55.

**Comment by markelliot** [comment]
Took #56 as a clean version of this.

**Pull request closed without merging**